pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -411,30 +411,37 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(pod *v1.Pod, podSchedulingC
 	if _, exists, _ := p.podBackoffQ.Get(pod); exists {
 		return fmt.Errorf("pod is already present in the backoffQ")
 	}
+
+	// Every unschedulable pod is subject to backoff timers.
+	p.backoffPod(pod)
+
+	// If there has been no move request since this pod failed scheduling, we can
```

Comment:
For both performance and readability reasons, I would suggest refactoring the following two `if` statements to:

```suggestion
if isPodUnschedulable(pod) {
	// If there has been no move request since this pod failed scheduling, we can
	// assume conditions have not changed and proceed normally.
	if podSchedulingCycle > p.moveRequestCycle {
		p.unschedulableQ.addOrUpdate(pod)
	} else {	
		// There was no move request, add the pod to the BackoffQ.
		err := p.podBackoffQ.Add(pod)
		if err != nil {
			return fmt.Errorf("Error adding pod %v to the backoff queue: %v", pod.Name, err)
		}
	}
	p.nominatedPods.add(pod, "")
	return nil
}
```

Reply:
Thanks, I've simplified the logic.